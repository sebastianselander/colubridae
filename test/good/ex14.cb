def foo(x: int) -> fn(int, string, bool, int) -> () {
    return \a b c d -> {
        printInt(a);
        printString(b);
        printBool(c);
        printInt(d+x);
    }
}

def main() {
    let f = foo(370);
    f(10,"yoo",true,50);
}


def printBool(b: bool) {
    if b {
        printString("true")
    } else {
        printString("false")
    }
}
