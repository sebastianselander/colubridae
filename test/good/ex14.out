10
yoo
true
420
